### PR TITLE
CLN: use IS64 instead of is_platform_32bit #36108

### LINKED ISSUE #36108: CLN: clean `pandas/compat.__init__.py`
Remove `is_platform_32bit()` from `pandas/compat.__init__.py`, use `IS64` instead to improve consistency.

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #36108
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/compat/__init__.py [hunk 1]
```diff
@@ -72,7 +62,7 @@ def is_platform_linux() -> bool:
     bool
         True if the running platform is linux.
     """
-    return sys.platform == "linux2"
+    return sys.platform == "linux"
```
Comment (VirosaLi) [inline pandas/compat/__init__.py:65]: It was changed to "linux" in Python 3.3

File: pandas/compat/__init__.py [hunk 2]
```diff
@@ -20,14 +19,6 @@
 IS64 = sys.maxsize > 2 ** 32
```
Comment (VirosaLi) [inline pandas/compat/__init__.py:21]: The LICENSES folder doesn't contain SIX. So I guess this comment is not valid anymore.

File: pandas/compat/__init__.py [hunk 3]
```diff
@@ -20,14 +19,6 @@
...
 
 
-# ----------------------------------------------------------------------------
-# functions largely based / taken from the six module
-
-# Much of the code in this module comes from Benjamin Peterson's six library.
-# The license for this library can be found in LICENSES/SIX and the code can be
-# found at https://bitbucket.org/gutworth/six
-
-
```
Comment (VirosaLi) [inline pandas/compat/__init__.py:30]: The LICENSES folder doesn't contain SIX. So I guess this comment is not valid anymore.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @VirosaLi very nice

**Pull request merged**